src/controllers/resize.ts

```diff
@@ -0,0 +1,62 @@
+import { Context } from 'koa'
+import { Controller, Ctx, Get, Params, Query } from 'amala'
+import { IPFS_TIMEOUT } from '../constants'
+import { Readable } from 'stream'
+import { badRequest, internal, notFound } from '@hapi/boom'
+import { getCachedResize, setCachedResize } from '../helpers/resizedImages'
+import { getContentType } from '../helpers/contentTypes'
+import ipfs, { getDataFromIPFS } from '../helpers/ipfs'
+import sharp from 'sharp'
+
+@Controller('/resize')
+export default class ResizeController {
+  @Get('/image/:cid')
+  async image(
+    @Ctx() ctx: Context,
+    @Params('cid') cid: string,
+    @Query('width') width?: number,
+    @Query('height') height?: number
+  ) {
+    try {
+      const contentType = await getContentType(cid)
+      if (contentType) ctx.set('Content-Type', contentType)
+      if (contentType && !contentType.startsWith('image/'))
```

Comment:
Should we do this check before `L22`?

Reply:
Resolved in [ca3c656](https://github.com/BigWhaleLabs/obss-ipfs-uploader/pull/12/commits/ca3c6560606fb19c0826ee71566006f5bf7eb33a)